services/oauth/v2/cp_connector_test.go

```diff
@@ -0,0 +1,94 @@
+package v2_test
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"syscall"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	mock_oauthV2 "github.com/rudderlabs/rudder-server/mocks/services/oauthV2"
+	v2 "github.com/rudderlabs/rudder-server/services/oauth/v2"
+)
+
+var _ = Describe("CpConnector", func() {
+	It("Test CpApiCall function to test success scenario", func() {
+		ctrl := gomock.NewController(GinkgoT())
+		mockHttpClient := mock_oauthV2.NewMockHttpClient(ctrl)
+		mockHttpClient.EXPECT().Do(gomock.Any()).Return(&http.Response{
+			StatusCode: http.StatusOK,
+			Body:       io.NopCloser(bytes.NewReader([]byte("test"))),
+		}, nil)
+
+		cpConnector := v2.ControlPlaneConnector{
+			Client: mockHttpClient,
+			Logger: logger.NewLogger().Child("ControlPlaneConnector"),
+		}
+		statusCode, respBody := cpConnector.CpApiCall(&v2.ControlPlaneRequestT{
+			Method:        "GET",
```

Comment:
Also, for similar occurances as well.
```suggestion
			Method:        http.MethodGet,
```